add documentation on enabling/disabling asynchronous-metric-log (#74450)
ClickHouse/ClickHouse: docs/en/operations/server-configuration-parameters/settings.md

Default: `120`

> ## asynchronous_metric_log

Reviewer: There is already a page for `asynchronous_metric_log` I think that the details in this PR should be added under [operations/system-tables/asynchronous_metric_log ](https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/operations/system-tables/asynchronous_metric_log.md).

Author: I added it here as we have a similar setting for `metric_log`. 

There is a system tables page: https://clickhouse.com/docs/en/operations/system-tables/metric_log

And a section in the global server setting to enable/disable them: https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#metric_log

However I need to add a link to the asynchronous_metric_log server settings in the system-table page.

Reviewer: Sure, I think it makes sense to add it in both places.

> It is enabled by default on production deployment. 

Reviewer: It's better to be more specific regarding what you mean by `production`, for example:
```suggestion
It's enabled by default on ClickHouse cloud deployments. 
```

Reviewer: FWIW, I pulled and ran `24.12` ClickHouse docker to double check. It doesn't seem to be specific to cloud/production environment and seems to be enabled by default. You might want to double check this one.
https://pastila.nl/?0052096c/71c6aa4f1def1a52bd26d3b6abb9166f#IDvTnHCu/HNvc48dkGlCKA==

Author: Good point. 

What I found is that when using the quick install approach the asynchronous metric log are not enabled by default: https://clickhouse.com/docs/en/install#quick-install

But when using Production deployment approach it is enabled: https://clickhouse.com/docs/en/install#available-installation-options

I can say it is enabled by default and add a note to specify that using quick install, it is not.

Reviewer: `nit:` Maybe it's just better to say that:

It's enabled by default on ClickHouse cloud deployments.
 
And be a bit generic about the `Note`. Somewhere along the lines:

For other environments, if it's not enabled by default, depending on how ClickHouse was installed or setup, you can manually enable or disable it as follows..

Default: `120`

## asynchronous_metric_log

Enabled by default on ClickHouse Cloud deployments.

> If the setting is not enabled by default on your environment, depending on how ClickHouse was installed, you can follow the instruction below to enable or disable it.

Reviewer: `nit:` It's not really a setting, it's a table.
```suggestion
If the table is not enabled by default on your environment, depending on how ClickHouse was installed, you can follow the instruction below to enable or disable it.
```

Reviewer: Other than this, the rest of it looks reasonable. Thank you for making the changes.

Reviewer: But I think it's fine, we are talking about settings in this section (similar to metric_log section).

**Enabling**

> To manually turn on asynchronous metric logs history collection [`system.asynchronous_metric_log`](../../operations/system-tables/asynchronous_metric_log.md), create `/etc/clickhouse-server/config.d/asynchronous_metric_log.xml` with the following content:

Reviewer 2: The same XML example is repeated 13 (!) times across this markdown file. In some places, the sub-tags (l. 48-55) are explained, in other cases (e.g. here), they are not and readers wonder what e.g. "reserved_size_rows" means ... it is not immediately obvious.

Instead of further copypasting the copypasta, we should
- document in a central place how `system.*_log`-type tables can be configured (sub-tags are common across these tables),
- check that really all parameters are documented (e.g. for example, the configuration template shipped with ClickHouse has a tag [<order_by>](https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.xml#L1100-L1108) which I don't see documented here,
- and have all `system.*-log` tables point to the central documentation.

The "central location" could be a new section at the end of this file.
